src/jvmMain/kotlin/io/beatmaps/api/users.kt

```diff
@@ -128,23 +133,43 @@ class UsersApi {
     data class List(val page: Long = 0, val api: UsersApi)
 
     @Location("/follow")
-    data class Follow(val Api: UsersApi)
+    data class Follow(val api: UsersApi)
+
+    @Location("/following/{user}/{page}")
+    data class Following(val user: Int, val page: Long = 0, val api: UsersApi)
+
+    @Location("/followedBy/{user}/{page}")
+    data class FollowedBy(val user: Int, val page: Long = 0, val api: UsersApi)
 }
 
+fun followData(uploaderId: Int, userId: Int?) = UserFollowData(
```

Comment:
Ok, ok, there's nothing wrong with this (except I don't think the way the ifs are written is that readable) but hear me out... What if...

```kotlin
fun followData(uploaderId: Int, userId: Int?): UserFollowData {
    val userFilter = Follows.userId eq uploaderId
    val followerFilter = Follows.followerId eq userId

    val userColumn = countWithFilter(userFilter)
    val followerColumn = if (userId != uploaderId) {
        Op.nullOp()
    } else {
        countWithFilter(followerFilter)
    }

    val bothColumn = if (userId == null || userId == uploaderId) {
        Op.nullOp()
    } else {
        countWithFilter(userFilter and followerFilter).greater(intLiteral(0))
    }

    return Follows
        .slice(userColumn, followerColumn, bothColumn).select {
            (Follows.userId eq uploaderId) or (Follows.followerId eq userId)
        }.single().let {
            UserFollowData(it[userColumn], it[followerColumn], it[bothColumn])
        }
}
```